### PR TITLE
Fix writing files with basic auth [sc-169330]

### PR DESCRIPTION
**Pull request by alexbourret**
Story details: https://app.shortcut.com/dataiku/story/169330

### CODE REVIEW

File: python-lib/robust_session.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import time
+import copy
```
Comment (Alexlandeau) [inline python-lib/robust_session.py:2]: 🧐 question: is it used ?

File: python-lib/robust_session.py [hunk 2]
```diff
@@ -24,12 +26,14 @@ def __init__(self, session=None, status_codes_to_retry=None, max_retries=1, base
         self.connection_kwargs = {}
         self.connection_library = None
         self.attempt_session_reset_on_403 = attempt_session_reset_on_403
+        self.default_headers = {}
 
-    def update_settings(self, session=None, status_codes_to_retry=None, max_retries=None, base_retry_timer_sec=None):
+    def update_settings(self, session=None, status_codes_to_retry=None, max_retries=None, base_retry_timer_sec=None, default_headers=None):
```
Comment (Alexlandeau) [inline python-lib/robust_session.py:31]: 🧑‍🍳 suggestion: 
```suggestion
    def update_settings(self, session=None, status_codes_to_retry=None, max_retries=None, base_retry_timer_sec=None, default_headers=self.default_headers):
```
and remove l.36

File: python-lib/sharepoint_client.py [hunk 3]
```diff
@@ -87,6 +87,11 @@ def __init__(self, config):
                 username=username,
                 password=password
             )
+            self.form_digest_value = get_form_digest_value(self.sharepoint_url, self.sharepoint_site, session=self.session)
+            default_headers = None
```
Comment (Alexlandeau) [inline python-lib/sharepoint_client.py:91]: 🧑‍🍳 suggestion:
```suggestion
            default_headers = {"X-RequestDigest": self.form_digest_value} if self.form_digest_value else None
```

File: python-lib/sharepoint_client.py [hunk 4]
```diff
@@ -947,21 +952,35 @@ def close():
...
-        logger.info("Getting form digest value")
+
+def get_form_digest_value(sharepoint_url, sharepoint_site, session=None, sharepoint_access_token=None):
+    def get_contextinfo_url():
+        return "https://{}/{}/_api/contextinfo".format(
+            sharepoint_url, sharepoint_site
+        )
+
+    logger.info("Getting form digest value")
+    if session is None:
```
Comment (Alexlandeau) [inline python-lib/sharepoint_client.py:963]: 🧑‍🍳 suggestion: `if session:`

**Review by Alexlandeau — CHANGES_REQUESTED** [review]
Bug was reproduced and is now fixed. Write is still bugged if the target location doesn't exist, but this issue will be fixed in https://github.com/dataiku/dss-plugin-sharepoint-online/pull/52

Minor changes requested

File: python-lib/robust_session.py (hunk 1 above)
Comment (alexbourret) [reply python-lib/robust_session.py]: 👍 Done in [b21dd28](https://github.com/dataiku/dss-plugin-sharepoint-online/pull/53/commits/b21dd28692865e6cd3a8d56c66849ffa347e8d65)

File: python-lib/robust_session.py (hunk 2 above)
Comment (alexbourret) [reply python-lib/robust_session.py:30]: My linter tells me I can't use `self` there...

File: python-lib/sharepoint_client.py (hunk 3 above)
Comment (alexbourret) [reply python-lib/sharepoint_client.py]: Done in [b21dd28](https://github.com/dataiku/dss-plugin-sharepoint-online/pull/53/commits/b21dd28692865e6cd3a8d56c66849ffa347e8d65)

File: python-lib/sharepoint_client.py (hunk 4 above)
Comment (alexbourret) [reply python-lib/sharepoint_client.py:961]: if not session ?

File: python-lib/robust_session.py (hunk 2 above)
Comment (Alexlandeau) [reply python-lib/robust_session.py:30]: Fine then, no problem !

**Review by Alexlandeau — APPROVED** [review]
LGTM !

**Pull request merged**